project-koku/koku
# [COST-3896] Add product/physicalCores and identity/TimeInterval to the AWS Line Items

## Jira Ticket

[COST-3896](https://issues.redhat.com/browse/COST-3896)

## Description

This change will add relevant AWS SWATCH columns to `aws_line_items`

## Testing

1. Checkout Branch
2. Restart Koku
3. Drop the `aws_line_items` trino table with `python scripts/cji_scripts/migrate_trino.py`
4. Run the following commands to load AWS test data
    ```
    make create-test-customer
    make load-test-customer-data test_source=aws
    ```
5. launch trino shell and check for the new columns.
    ```
     trino --server 127.0.0.1:8080 --catalog hive --schema=org1234567 --user admin --debug

     SHOW COLUMNS FROM aws_line_items LIKE  '%interval';
    ```
    1. You should see the `identity_timeinterval` column.
    2. If you have the changes from the nise pr linked in the notes section, you should see also the `product_physicalcores` column when you run the command below otherwise zero columns will be returned.
    ```
     SHOW COLUMNS FROM aws_line_items LIKE  '%cores';
    ```


## Notes

...
Requires https://github.com/project-koku/nise/pull/453 for appropriate test data
